plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/common/ColorField.tsx

```diff
@@ -1,20 +1,43 @@
+import { Icon } from "@web/sharedComponents";
 import { styled } from "@web/theme";
+import { ChangeEvent, useState } from "react";
 
-import { FieldTitle, FieldValue, FieldWrapper, TextInput } from "./styled";
+import { FieldTitle, FieldValue, FieldWrapper, TextInput } from "../commonStyles";
 
 type Props = {
   title: string;
   titleWidth?: number;
   color?: string;
+  onChange?: (color: string) => void;
 };
 
-const ColorField: React.FC<Props> = ({ title, titleWidth, color }) => {
+function isValidColor(color: string) {
+  return CSS.supports("color", color);
+}
+
+const ColorField: React.FC<Props> = ({ title, titleWidth, color, onChange }) => {
+  const [text, setText] = useState(color);
+  const [selectedColor, setSelectedColor] = useState(color);
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setText(e.target.value);
+    setSelectedColor(() => {
+      const result = isValidColor(e.target.value) ? e.target.value : "";
+      onChange?.(result);
+      return result;
+    });
+  };
```

Comment:
Have you tested this by manually typing in a value?
If I understand the code correctly, if I want to type "red", and I type the first r, wouldn't this revert back to ""? So I'd never be able to manually type in a color, only copy/paste

Reply:
I have tested it and it works, can you please test it as well.
Please note that the color field has two states one for the text and the other one for the selected color.